tests/generation/test_export.py

```diff
@@ -30,15 +33,21 @@
 )
 @is_inferentia_test
 @requires_neuronx
-def test_decoder_export(export_decoder_id, batch_size, sequence_length, num_cores, auto_cast_type):
-    model = NeuronModelForCausalLM.from_pretrained(
-        export_decoder_id,
-        export=True,
-        batch_size=batch_size,
-        sequence_length=sequence_length,
-        num_cores=num_cores,
-        auto_cast_type=auto_cast_type,
-    )
+@pytest.mark.parametrize("local", [True, False], ids=["local", "from_hub"])
+def test_decoder_export(local, export_decoder_id, batch_size, sequence_length, num_cores, auto_cast_type):
+    export_kwargs = {
+        "batch_size": batch_size,
+        "sequence_length": sequence_length,
+        "num_cores": num_cores,
+        "auto_cast_type": auto_cast_type,
+    }
+    if local:
+        with TemporaryDirectory() as model_path:
```

Comment:
I recently discovered the tmpdir pytest that can be useful as well.

Reply:
Indeed, thanks ! I can't use it here because I only need it if local=true